### ISSUE TITLE
ModuleAPI, Add methods to check and change user profiles and threepids

### ISSUE THREAD

**Issue by pixelplanetdev** [matrix-org/synapse#11746]
There is [get_profile_for_user](https://github.com/matrix-org/synapse/blob/9ec46d6231c267a1aa25c96b5d86822c7da8fca9/synapse/module_api/__init__.py#L473)
There is [get_threepids_for_user](https://github.com/matrix-org/synapse/blob/9ec46d6231c267a1aa25c96b5d86822c7da8fca9/synapse/module_api/__init__.py#L486)
But there are no methods to set them.
We can just set them when registering a new user.... but we can't register a new user on each login.

## Background

I am writing an authentication module for my website [here](https://github.com/pixelplanetdev/ppfun-matrixbridge/blob/master/synapse-auth-module/ppfun-auth.py) similar to the LDAP3 module and if someone logs in with a user a second time, it checks if the emails are still the same, and if not, updates it in the matrix user.

This looks like this currently:

```py
        threepid_dict_list = await self.api.get_threepids_for_user(user_id)
        for threepid_dict in threepid_dict_list:
            if threepid_dict['address'] == email:
                logger.info('User %s has same email as in ppfun. Nice', user_id)
                return False
        logger.info('User %s did not have email set, setting it to %s', user_id, email)
        current_time = self.api._hs.get_clock().time_msec()
        threepid_dict = {
            "medium": "email",
            "address": email,
            "validated_at": current_time,
        }
        registration_handler = self.api._hs.get_registration_handler()
        await registration_handler._register_email_threepid(user_id, threepid_dict, None)
        return True
```

...so i wade through _privateMethods and Atributes... which i am surprised that i can even do that?
Like `self.api._hs.get_registration_handler()._register_email_threepid(user_id, threepid_dict, None)`
I also have to get the server name with `self.api._server_name`.

**Comment by babolivier** [comment]
> ...so i wade through _privateMethods and Atributes... which i am surprised that i can even do that?

Yes, currently modules aren't sandboxed to prevent them from accessing properties they shouldn't; however as you seem to imply doing that is very not recommended because it can break at any point without warning.

>  I also have to get the server name with `self.api._server_name`.

Gah, I thought we already exposed the server name through the module API, but apparently we don't, sorry about this oversight.

> There is get_profile_for_user
> There is get_threepids_for_user
> But there are no methods to set them.

Yep, methods to set profile and 3pids definitely have their place in the `ModuleApi`, we just haven't got round to doing that yet.
